python_modules/dagster/dagster/_core/pipes/utils.py

```diff
@@ -340,8 +327,33 @@ def _messages_thread(
                     break
             time.sleep(1)
 
+    def _logs_thread(
+        self,
+        params: PipesParams,
+        is_task_complete: Event,
+    ) -> None:
+        # Start once we have received the opened message from the external process
+        while self.opened_payload is None:
+            time.sleep(0.1)
+        log_params = {**params, **self.opened_payload}
+        for log_reader in self.log_readers:
+            log_reader.start(log_params, is_task_complete)
+        while True:
+            if is_task_complete.is_set():
+                break
+            else:
+                time.sleep(1)
+
+        # In cases where we are forwarding logs, in some cases the logs might not be written out until
+        # after the run completes. We wait for them to exist.
+        log_wait_start = datetime.datetime.now()
+        while (datetime.datetime.now() - log_wait_start).seconds <= WAIT_FOR_LOGS_TIMEOUT and (
+            any((not log_reader.is_ready(log_params) for log_reader in self.log_readers))
+        ):
```

Comment:
under what circumstances would `is_ready` never become true?

Reply:
`is_ready` is left to the individual log reader to implement, but in the case of DBFS, it would never be true if the targeted file on DBFS never ends up existing or if it briefly existed but was deleted between polling calls.